src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -1269,10 +1269,23 @@ public function get_index( $request ) {
 		);
 
 		$response = new WP_REST_Response( $available );
-		$response->add_link( 'help', 'https://developer.wordpress.org/rest-api/' );
-		$this->add_active_theme_link_to_index( $response );
-		$this->add_site_logo_to_index( $response );
-		$this->add_site_icon_to_index( $response );
+
+		$fields = isset( $request['_fields'] ) ? $request['_fields'] : '';
+		$fields = wp_parse_list( $fields );
+		if ( empty( $fields ) ) {
+			$fields[] = '_links';
+		}
+
+		if ( $request->has_param( '_embed' ) ) {
+			$fields[] = '_embedded';
+		}
+
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$response->add_link( 'help', 'https://developer.wordpress.org/rest-api/' );
+			$this->add_active_theme_link_to_index( $response );
+			$this->add_site_logo_to_index( $response );
+			$this->add_site_icon_to_index( $response );
```

Comment:
`$this->add_site_icon_to_index( $response );` is removed in trunk, possibly there are more changes, patch failed for all three files 😶

Reply:
@OlaIola I am sorry, I don't understand that feedback. I am still seeing it in trunk. https://github.com/WordPress/wordpress-develop/blob/f933aa3616aa8d49bdd67886e4af287579093c25/src/wp-includes/rest-api/class-wp-rest-server.php#L1277

Reply:
Forgive me, @spacedmonkey, I've checkout master automatically 🤦
I checked the patch, it applies correctly, but there is a difference, when site_logo is 0 it returns an empty array instead of this value.
/wp-json/?_fields=site_logo
/wp-json/?_fields=site_logo,_links - this way it is present